### PR TITLE
Mutable Shared State

### PR DESCRIPTION
**Pull request by VictorXLR**
The server can now modify game state
with the addition of a sharable mutex state, which can be used to modify data, we can create Game objects now from the data sent from the post request at endpoint `game`

### CODE REVIEW

File: server/src/main.rs [hunk 1]
```diff
@@ -25,8 +27,9 @@ async fn game_count(model: web::Data<Model>) -> impl Responder {
 
 /// POST /game/create This
 /// function will be called from a post request
-///
-async fn post_game(form: web::Form<FormData>) -> impl Responder {
+async fn post_game(form: web::Form<FormData>, model: web::Data<Mutex<Model>>) -> impl Responder {
+    let mut g_model = model.lock().unwrap();
+    g_model.add_game(game::init(&form.game_name));
```
Comment (Chadtech) [inline server/src/main.rs]: What is `g_` in `g_model`?

Could `model` be renamed like `mutex` or `data` and then `let mut model = mutex.loc().unwrap()` or something? I dont know what names make sense, I am just saying you can free up `model` if you name the prior thing something else.

**Review by Chadtech — APPROVED** [review]
Looks good!

File: server/src/main.rs (hunk 1 above)
Comment (VictorXLR) [reply server/src/main.rs]: I renamed it to `data` will make a pull request now

Comment (VictorXLR) [reply server/src/main.rs]: I also renamed the app_data from `g_model` to global state

**Pull request merged**